lib/kino/fs.ex

```diff
@@ -0,0 +1,67 @@
+defmodule Kino.FS do
+  @moduledoc """
+  Provides access to notebook files.
+  """
+
+  @doc """
+  Accesses notebook file with the given name and returns a local path
+  to ead its contents from.
+
+  This invocation may take a while, in case the file is downloaded
+  from a URL and is not in the cache.
+
+  > #### File operations {: .info}
+  >
+  > You should treat the file as read-only. To avoid unnecessary
+  > copies the path may potentially be pointing to the original file,
+  > in which case any write operations would be persisted. This
+  > behaviour is not always the case, so you should not rely on it
+  > either.
+  """
+  @spec file_path(String.t()) :: String.t()
+  def file_path(name) when is_binary(name) do
+    case Kino.Bridge.get_file_entry_path(name) do
+      {:ok, path} ->
+        path
+
+      {:error, message} when is_binary(message) ->
+        raise message
+
+      {:error, reason} when is_atom(reason) ->
+        raise "failed to access file path, reason: #{inspect(reason)}"
+    end
+  end
+
+  @typep file_spec ::
+           %{
+             type: :local,
+             path: String.t()
+           }
+           | %{
+               type: :url,
+               url: String.t()
+             }
+           | %{
+               type: :s3,
+               bucket_url: String.t(),
+               region: String.t(),
+               access_key_id: String.t(),
+               secret_access_key: String.t(),
+               key: String.t()
+             }
```

Comment:
Just an update: we started working on the `FSS` library. However, I believe only `Kino` should depend on `FSS`, because there is no guarantee Livebook and Kino will have the same `FSS` version. So Kino.Bridge can keep on using the format above and the function below will pattern match and convert it to the relevant FSS entries.

FSS will look like this:

    %FSS.S3.Entry{
      url: string,
      config: %FSS.S3.Config{
        access_key_id: ...
        ...
      }
    }

    %FSS.Local.Entry{
      path: string
    }

    %FSS.HTTP.Entry{
      url: string,
      config: %FSS.HTTP.Config{
        headers: [...]
      }
    }

Reply:
Sounds good!